File: model/article.go
package model

import "gorm.io/gorm"

type Article struct {
	gorm.Model
	Title    string    `gorm:"size:255;not null" json:"title"`
	Content  string    `gorm:"type:text;not null" json:"content"`
	UserID   uint      `gorm:"not null" json:"user_id"`
	Media    []Media   `json:"media"`
	Comments []Comment `json:"comments" gorm:"foreignKey:ArticleID;constraint:OnDelete:CASCADE;"`
}

type Media struct {
	gorm.Model
	ArticleID uint   `gorm:"index;not null" json:"article_id"` // 外键
	Type      string `json:"type"`
	URL       string `json:"url"`
	ObjectKey string `json:"objectKey"`
	Size      int64  `json:"size"`
	Duration  int    `json:"duration"`
}

type Comment struct {
	gorm.Model
	ArticleID uint      `gorm:"index;not null" json:"article_id"`
	UserID    uint      `gorm:"not null" json:"user_id"`
	Content   string    `gorm:"type:text;not null" json:"content"`
	ParentID  *uint     `json:"parent_id"`        // nil 表示顶级评论
	Replies   []Comment `json:"replies" gorm:"-"` // 在 service 层组装，不持久化到 DB
}

File: web/src/api/article.ts
import request from '@/utils/request'

export interface Media {
  id: number
  type: string
  url: string
  objectKey: string
  size: number
  duration: number
}

export interface Article {
  id: number
  title: string
  content: string
  user_id: number
  media: Media[]
  created_at: string
  updated_at: string
}

export function articleList() {
  return request.get<{ articles: Article[] }>('/articles')
}

export function articleDetail(id: number) {
  return request.get<{ article: Article }>(`/articles/${id}`)
}

export function createArticle(formData: FormData) {
  return request.post<{ msg: string; article: Article }>('/articles/create', formData, {
    headers: { 'Content-Type': 'multipart/form-data' },
  })
}

export function deleteArticle(id: number) {
  return request.delete<{ msg: string }>(`/articles/${id}`)
}

export function getComments(articleId: number) {
  return request.get<any[]>(`/articles/${articleId}/comments`)
}

export function createComment(articleId: number, data: { content: string; parent_id?: number }) {
  return request.post(`/articles/${articleId}/comments`, data)
}

export function deleteComment(commentId: number) {
  return request.delete(`/articles/comments/${commentId}`)
}
